### PR TITLE
perf(solution): add non sso tab in 'add capability'

### PR DESCRIPTION
**Pull request by KennethBWSong**

### CODE REVIEW

File: packages/fx-core/src/plugins/solution/fx-solution/v2/executeUserTask.ts [hunk 1]
```diff
@@ -297,9 +299,39 @@ export async function addCapability(
...
+        SolutionError.InvalidInput,
+        getLocalizedString("core.addSsoFiles.canNotAddTabWhenSsoNotEnabled"),
+        SolutionSource
+      );
+      return err(e);
+    }
+
+    if (toAddTabNonSso) {
+      const index = capabilitiesAnswer.indexOf(TabNonSsoItem.id);
+      capabilitiesAnswer.splice(index);
```
Comment (1yefuwang1) [inline packages/fx-core/src/plugins/solution/fx-solution/v2/executeUserTask.ts:330]: `Array.splice()` here will remove all items within range [index, capabilitiesAnswer.length).

Comment (1yefuwang1) [reply packages/fx-core/src/plugins/solution/fx-solution/v2/executeUserTask.ts:330]: Besides, I'm a little bit confused about why you want to remove `TabNonSsoItem.id` and manually add `TabOptionItem.id` here when `toAddTabNonSso` is true.

File: packages/fx-core/src/plugins/solution/fx-solution/v2/executeUserTask.ts [hunk 2]
```diff
@@ -398,11 +482,37 @@ export async function addCapability(
       pluginNamesToScaffold.add(ResourcePluginsV2.FrontendPlugin);
       if (!alreadyHasBot && !alreadyHasME) {
         pluginNamesToArm.add(ResourcePluginsV2.BotPlugin);
+
+        if (isAadManifestEnabled() && alreadyHasSso) {
+          const createAuthFilesRes = await createAuthFiles(inputsNew, false, true, true);
+          if (createAuthFilesRes.isErr()) {
```
Comment (SLdragon) [inline packages/fx-core/src/plugins/solution/fx-solution/v2/executeUserTask.ts:488]: Seems return error logic are all same, can we extract to a function to make it simple

File: packages/fx-core/src/plugins/solution/fx-solution/v2/executeUserTask.ts [hunk 3]
```diff
@@ -876,3 +1004,58 @@ export async function addSso(
 
   return ok(undefined);
 }
+
+export async function createAuthFiles(
+  input: Inputs,
+  needTab: boolean,
+  needBot: boolean,
+  isVsProject = false
```
Comment (SLdragon) [inline packages/fx-core/src/plugins/solution/fx-solution/v2/executeUserTask.ts:1012]: isVsProject is not used, we can add a todo for it

File: packages/fx-core/src/plugins/solution/fx-solution/v2/executeUserTask.ts (hunk 1 above)
Comment (KennethBWSong) [reply packages/fx-core/src/plugins/solution/fx-solution/v2/executeUserTask.ts:330]: Both tab and non sso tab activates fe plugin, and fe plugin will handle the logic whether to scaffold sso or non sso template. What we need to do is to add a tab. By remove the non sso tab and add the tab option, we can keep the logic below same as current :
image

File: packages/fx-core/src/plugins/solution/fx-solution/v2/executeUserTask.ts (hunk 2 above)
Comment (KennethBWSong) [reply packages/fx-core/src/plugins/solution/fx-solution/v2/executeUserTask.ts:488]: Sure, will do

File: packages/fx-core/src/plugins/solution/fx-solution/v2/executeUserTask.ts (hunk 1 above)
Comment (1yefuwang1) [reply packages/fx-core/src/plugins/solution/fx-solution/v2/executeUserTask.ts:330]: OK, that's a little bit tricky.

**Review by SLdragon — APPROVED** [review]

**Review by 1yefuwang1 — APPROVED** [review]

**Pull request merged**